sycl/test/kernel-and-program/program_cache.cpp

```diff
@@ -16,11 +16,19 @@ class Functor {
   }
 };
 
+namespace pi = cl::sycl::detail::pi;
```

Comment:
It seems new kernel_cache.cpp  covers all test cases in program_cache. What about removing program_cache then?